x-pack/plugins/lens/public/datasources/form_based/operations/definitions/filters/filter_popover.tsx

```diff
@@ -36,9 +36,18 @@ export const FilterPopover = ({
   triggerClose: () => void;
 }) => {
   const inputRef = React.useRef<HTMLInputElement>();
+  const [localFilter, setLocalFilter] = useState(() => filter);
```

Comment:
What's the purpose of using the callback approach to initialize the state?

```suggestion
  const [localFilter, setLocalFilter] = useState(filter);
```

Reply:
I just prefer to use it this way because in some cases it can be more performant so sometimes I do it as a default (doesn't harm anything) 
https://legacy.reactjs.org/docs/hooks-reference.html#lazy-initial-state